Packs/SymantecCloudSOC/Integrations/SymantecCloudSOC/SymantecCloudSOC.py

```diff
@@ -0,0 +1,431 @@
+                      f"for log type: {log_type}")
+        demisto.debug(f"SymantecEventCollector: Received {len(log_events)} events for log type {log_type}")
+        events.extend(log_events)
+
+    demisto.debug(f"SymantecEventCollector: Returning {len(events)} events in total")
+    return new_last_run, events
+
+
+def create_client_with_authorization(base_url: str, verify_certificate: bool,
+                                     proxy: bool, key_id: str, key_secret: str) -> Client:
+    """
+    Creates a client with basic access authentication.
+    Args:
+        base_url (str): Symantec Cloud SOC base URL.
+        verify_certificate (bool): Whether the request should verify the SSL certificate.
+        proxy (bool): Whether to run the integration using the system proxy.
+        key_id (str): Symantec Cloud SOC key ID.
+        key_secret (str): Symantec Cloud SOC key secret.
+    Returns:
+        Client: Client class to interact with Symantec Cloud SOC service API.
+    """
+    credentials = f'{key_id}:{key_secret}'
+    encoded_credentials = base64.b64encode(credentials.encode()).decode()
+    headers = {
+        'Authorization': f'Basic {encoded_credentials}',
+        'X-Elastica-Dbname-Resolved': 'True'
+    }
+    return Client(base_url=base_url,
+                  verify=verify_certificate,
+                  headers=headers,
+                  proxy=proxy)
+
+
+def get_first_fetch_time(first_fetch_from_params: str = None) -> tuple[str, str]:
+    """
+    Gets a first_fetch_time arg for investigate logs and detect incidents logs.
+    Args:
+        first_fetch_from_params (str): first_fetch parameter from XSIAM.
+    Returns:
+        first_fetch_str (str): first fetch.
+        first_fetch_time_investigate_str (str): first fetch for investigate log type.
+    """
+    # How much time before the first fetch to retrieve events
+    first_fetch = first_fetch_from_params or '3 days'
+    first_fetch_time: datetime = arg_to_datetime(arg=first_fetch,
+                                                 arg_name='First fetch time',
+                                                 required=True)  # type: ignore[assignment]
+    first_fetch_str = first_fetch_time.strftime(DATE_FORMAT_SYMANTEC)
+    six_month_ago: Optional[datetime] = dateparser.parse('6 months', settings={'TIMEZONE': 'UTC'})
+    demisto.debug(f'SymantecEventCollector: first fetch time: {first_fetch_time}')
+    # For investigate app type the created_timestamp must be less than 6 months.
+    first_fetch_time_investigate: datetime = first_fetch_time
+    if six_month_ago and first_fetch_time_investigate <= six_month_ago:
+        first_fetch_time_investigate = dateparser.parse('180 day', settings={'TIMEZONE': 'UTC'})  # type: ignore[assignment]
+        first_fetch_time_investigate += timedelta(minutes=5)
+    first_fetch_time_investigate_str = first_fetch_time_investigate.strftime(DATE_FORMAT_SYMANTEC)
+    return first_fetch_str, first_fetch_time_investigate_str
```

Comment:
You cannot retrieve investigate logs that are older than 180 days. Therefore, if setting a first fetch that is more than 180 days, for investigate logs it will be a maximum of 180 days.